src/shapes/path.class.js

```diff
@@ -66,7 +71,24 @@
       if (!this.path) {
         return;
       }
-      fabric.Polyline.prototype._setPositionDimensions.call(this, options);
+      fabric.Polyline.prototype._setPositionDimensions.call(this, options || {});
+    },
+
+    /**
+     * @private
+     */
+    _set: function (key, value) {
+      if (key === 'path') {
+        var left = this.left,
+            top = this.top;
+        this._setPath(value);
+        this.left = left;
+        this.top = top;
+        this.setCoords();
```

Comment:
this is a bit of a stretch for a setter and is also a breaking change because is radically changing how the app behaves for who uses set today
This is not wrong but should be an utility on its own.

Something like `path.updatePath` that can do exactly

```
        var left = this.left,
            top = this.top;
        this._setPath(value);
        this.left = left;
        this.top = top;
        this.setCoords();
```

Reply:
You are right. It was breaking.
Now it isn't a2373e2.
So to preserve the position one can easily do the following:
```js
        var left = path.left, top = path.top;
        path.set('path', value);
        path.left = left;
        path.top = top;
```